File: data-structure-week-1/valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
            int n=s.length();
        int m=t.length();
        if(n!=m)return false;
        map<char,int>mp;
        for(int i=0;i<n;i++){
            mp[s[i]]++;
        }
        
        for(int i=0;i<n;i++){
            if(mp[t[i]])
                mp[t[i]]--;
            else
                return false;
        }
        return true;
    }
};

File: data-structure-week-1/valid-anagram/valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    
    if(s.length !== t.length) {
        return false;
    }
    let sMap = new Map();
    
    for(let i=0; i<s.length; i++) {
        sMap.set(s.charAt(i), sMap.has(s.charAt(i)) ? sMap.get(s.charAt(i)) + 1 : 1 );
    }
    
    for(let i=0; i<t.length; i++) {
        if(sMap.has(t.charAt(i))) {
            sMap.set(t.charAt(i), sMap.get(t.charAt(i)) - 1);
            if(sMap.get(t.charAt(i)) < 0) {
                return false;
            }
        } else {
            return false;
        }
    }
    
    return true;    
};

